tests/unit/dccs/test_aws_iot.py

```diff
@@ -0,0 +1,615 @@
+        """
+        Method to test the validation of create_relationship method of class AWSIoT.
+        :return: None
+        """
+
+        # Monkey patch the constructor of MqttDccComms
+        MqttDccComms.__init__ = mocked_init_mqtt_dcc_comms
+
+        # Creating the mocked MqttDccComms object
+        mocked_mqtt_dcc_comms = MqttDccComms()
+
+        aws = AWSIoT(mocked_mqtt_dcc_comms, enclose_metadata=True)
+
+        # Register the edge
+        registered_entity = aws.register(self.edge_system)
+
+        with self.assertRaises(TypeError):
+            # Creating the parent-child relationship between Edge-System and Edge-System
+            aws.create_relationship(registered_entity, registered_entity)
+
+    def test_implementation_get_entity_hierarchy(self):
+        """
+        Method to test the implementation of create_relationship method of class AWSIoT.
+        :return: None
+        """
+
+        # Monkey patch the constructor of MqttDccComms
+        MqttDccComms.__init__ = mocked_init_mqtt_dcc_comms
+
+        # Creating the mocked MqttDccComms object
+        mocked_mqtt_dcc_comms = MqttDccComms()
+
+        aws = AWSIoT(mocked_mqtt_dcc_comms, enclose_metadata=True)
+
+        # Register the edge
+        registered_entity = aws.register(self.edge_system)
+
+        #  Creating test Metric
+        test_metric = Metric(
+            name="Test_Metric",
+            unit=None,
+            interval=10,
+            aggregation_size=2,
+            sampling_function=sampling_function
+        )
+
+        registered_metric = aws.register(test_metric)
+
+        # Creating the parent-child relationship
+        aws.create_relationship(registered_entity, registered_metric)
+
+        # Getting the parent child relationship array from registered metric
+        entity_hierarchy = aws._get_entity_hierarchy(registered_metric)
+
+        self.assertSequenceEqual([registered_entity.ref_entity.name, registered_metric.ref_entity.name],
+                                 entity_hierarchy, "Check the implementation of _get_entity_hierarchy")
+
+    def test_validation_get_entity_hierarchy(self):
+        """
+        Method to test the validation of _get_entity_hierarchy method of class AWSIoT.
+        :return: None
+        """
+
+        # Monkey patch the constructor of MqttDccComms
+        MqttDccComms.__init__ = mocked_init_mqtt_dcc_comms
+
+        # Creating the mocked MqttDccComms object
+        mocked_mqtt_dcc_comms = MqttDccComms()
+
+        aws = AWSIoT(mocked_mqtt_dcc_comms, enclose_metadata=True)
+
+        # Creating test Metric
+        test_metric = Metric(
+            name="Test_Metric",
+            unit=None,
+            interval=10,
+            aggregation_size=2,
+            sampling_function=sampling_function
+        )
+
+        # Checking whether implementation raising the TypeError for invalid input
+        with self.assertRaises(TypeError):
+            aws._get_entity_hierarchy(test_metric)
+
```

Comment:
Add another test case for get_entity_hierarchy():
"Test case to check get_entity_entity_hierarchy() for RegisteredEdgeSystem->RegisteredDevice->RegisteredMetric"